### PR TITLE
experimentation: Add buildbreaker script and workflow

### PR DESCRIPTION
**Pull request by slarse**
This is a first step toward #478, and also related to #440

The `sorald.buildbreaker` script is a crude prototype of breaking the build when Sorald finds a repairable violation. It works as follows.

1. Mine violations for all handled rules
2. Perform targeted repair of the mined violations
    - This will in the future allow us to only select violations that are within edited parts of the code!
3. If Sorald succesfully repairs any of the found violations, exit non-zero. Otherwise, exit zero.

This PR also adds a GHA workflow that first fetches the latest version of Sorald from Maven Central, and then runs the buildbreaker script. Currently, a lot of stuff is hard-coded (such as the version of Sorald to use, and the source path to analyze). That will later be input for the GitHub Action.

* Example of running the workflow when there are repairable violations (i.e. **failure**): https://github.com/SpoonLabs/sorald/runs/2342788710#step:7:8
    - Note: I've since changed the "error" message to `Repairable violations found: <violations>`
* Example of running the workflow when there are no repairable violations (i.e. **success**): https://github.com/SpoonLabs/sorald/runs/2342832172#step:7:8

This is all very much a WIP, and things are bound to be tweaked, but I think it's a good first attempt that's ready to be merged into the repo. We'll iterate on this until satisfied, and then create a GitHub Action of it.

### CODE REVIEW

**Comment by slarse** [comment]
As I've already progressed so far with #486, perhaps we should not merge these scripts. They were a good starting point for my prototyping, but I don't think we'll need them now.

**Review by khaes-kth — APPROVED** [review]
Looks very good to me!

**Comment by khaes-kth** [comment]
@slarse Do you think we should wait until #486 is ready and merge that one instead?

**Comment by slarse** [comment]
@khaes-kth Oh sorry, did you miss my comment before reviewing? Yeah, I'm not sure of the value of these scripts given that the end-goal is to encompass them into the GitHub Action anyway, and I've progressed so far with the prototype.

**Comment by khaes-kth** [comment]
Great. Then, we'll wait for it.

BTS, Why shouldn't we close this PR without merge?

**Comment by slarse** [comment]
> BTS, Why shouldn't we close this PR without merge?

I just keep it open for visibility at the moment.

**Comment by slarse** [comment]
So, this is now over and done with as #486 has come to the point where it fulfills the same role. We don't need these scripts anymore.

**Pull request closed without merging**
